src/views/EditView.jsx

```diff
@@ -1,3 +1,63 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { useBackend } from "../state/use-backend";
+import { updateUser } from "../state/update-user";
+
+const EditView = ({ match }) => {
+  const initialFormState = [""];
+  const { uname } = match.params;
+  const { data } = useBackend(uname);
+  const [edit, setEdit] = useState([]);
+  const user = { ...data };
+
+  const onChange = event => {
+    const { name, value } = event.target;
+    setEdit({ ...edit, [name]: value });
+    console.log(edit);
+  };
+
+  const onSubmit = event => {
+    event.preventDefault();
+    if (edit.length !== 0) {
+      updateUser(uname, edit);
+      setEdit(initialFormState);
+    } else alert("Oops! Looks like you forgot to add changes.");
+  };
+
+  return (
+    <div>
+      <h2 className="edit-profile">Edit user: {match.params.uname}</h2>
+      <h4 className="edit-profile">Name: {user.name}</h4>
+      <h4 className="edit-profile">Surname: {user.surname}</h4>
+      <form onSubmit={onSubmit}>
+        <label className="label__edit">Name: </label>
+        <input className="input__edit" name="name" onChange={onChange}></input>
```

Comment:
inputs should implement controlled states with bot `value` and `onChange` bound to a local state.
Let's make an effort and find nice tutorials regarding form handling.

Reply:
Have spent hours trying to figure this out, but haven't been able to correct it so that the value sends the original through if not changed. Hope we can merge this so that someone else can work on it too.